lib/mutant/mutator/node/send.rb

```diff
@@ -117,6 +118,22 @@ def emit_receiver_selector_mutations
             .each(&method(:emit_selector))
         end
 
+        # Emit mutation for `#dig`
+        #
+        # - Mutates `foo.dig(a, b)` to `foo.fetch(a).dig(b)`
+        # - Mutates `foo.dig(a)` to `foo.fetch(a)`
+        #
+        # @return [undefined]
+        def emit_dig_mutation
+          return unless selector.equal?(:dig) && arguments.any?
+
+          fetch_mutation = s(:send, receiver, :fetch, arguments.first)
```

Comment:
One reason I think it's better is that it replaces method calls with language features. Instead of calling methods to find the head and tail of the list it's using pattern matching to extract the values. I tend to prefer using built-in syntax over method calls.